### PR TITLE
Fix #2198: Allow non-existent classpath entries

### LINKED ISSUE #2198: Regression: fastOptJS fails with Illegal classpath entry: <non existent path> after upgrading from 0.6.5 to 0.6.6
Here is the error message when I run `sbt clean fastOptJS`

```scala
java.lang.RuntimeException: Illegal classpath entry: /path-to-project/Web/
target/web/classes/main
        at scala.sys.package$.error(package.scala:27)
        at org.scalajs.sbtplugin.ScalaJSPluginInternal$$anonfun$org$scalajs$sbtplugin$ScalaJSPluginIntern
al$$collectFromClasspath$1.apply(ScalaJSPluginInternal.scala:322)
        at org.scalajs.sbtplugin.ScalaJSPluginInternal$$anonfun$org$scalajs$sbtplugin$ScalaJSPluginIntern
al$$collectFromClasspath$1.apply(ScalaJSPluginInternal.scala:309)
        at scala.collection.immutable.List.foreach(List.scala:318)
        at org.scalajs.sbtplugin.ScalaJSPluginInternal$.org$scalajs$sbtplugin$ScalaJSPluginInternal$$coll
ectFromClasspath(ScalaJSPluginInternal.scala:309)
        at org.scalajs.sbtplugin.ScalaJSPluginInternal$$anonfun$37.apply(ScalaJSPluginInternal.scala:464)
        at org.scalajs.sbtplugin.ScalaJSPluginInternal$$anonfun$37.apply(ScalaJSPluginInternal.scala:463)
        at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
        at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
        at sbt.std.Transform$$anon$4.work(System.scala:63)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.Execute.work(Execute.scala:235)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
        at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
        at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
```

note that `show Web/compile:fullClasspath` does contain non-existent `/path-to-project/Web/target/web/classes/main`

@sjrd suggest a fix by adding 
```scala
fullClasspath in (Web, Compile) ~= { _.filter(_.data.exists) }
```
works though.

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

**Comment by gzm0** [comment]
Review by @sjrd

File: sbt-plugin-test/build.sbt [hunk 1]
```diff
@@ -11,7 +11,10 @@ val versionSettings = Seq(
 
 val baseSettings = versionSettings ++ Seq(
   libraryDependencies +=
-    "org.scala-js" %% "scalajs-jasmine-test-framework" % scalaJSVersion % "test"
+    "org.scala-js" %% "scalajs-jasmine-test-framework" % scalaJSVersion % "test",
+
+  // Test that non-existent classpath entries are allowed - #2198
+  fullClasspath += file("./non-existent-directory-please-dont-ever-create-this")
```
Comment (sjrd) [inline sbt-plugin-test/build.sbt]: This should not be a relative path.

**Comment by sjrd** [comment]
That's all

File: sbt-plugin-test/build.sbt (hunk 1 above)
Comment (gzm0) [inline sbt-plugin-test/build.sbt]: Why not? Isn't this relative to the sbt root?

Comment (sjrd) [inline sbt-plugin-test/build.sbt]: Maybe, but by accident. 

The only place where sbt guarantees the meaning of `.` is in the initial setting of projects' base directories as in `.in(file("."))`. Elsewhere it's undefined.

Comment (gzm0) [inline sbt-plugin-test/build.sbt]: Learned something today :)

**Comment by gzm0** [comment]
Updated

**Comment by sjrd** [comment]
LGTM

File: sbt-plugin-test/build.sbt [hunk 2]
```diff
@@ -11,7 +11,11 @@ val versionSettings = Seq(
 
 val baseSettings = versionSettings ++ Seq(
   libraryDependencies +=
-    "org.scala-js" %% "scalajs-jasmine-test-framework" % scalaJSVersion % "test"
+    "org.scala-js" %% "scalajs-jasmine-test-framework" % scalaJSVersion % "test",
+
+  // Test that non-existent classpath entries are allowed - #2198
+  fullClasspath +=
+    file(root.base / "non-existent-directory-please-dont-ever-create-this")
```
Comment (sjrd) [inline sbt-plugin-test/build.sbt]:
```
sbt-plugin-test/build.sbt:18: error: recursive value root needs type
    file(root.base / "non-existent-directory-please-dont-ever-create-this")
         ^
```

**Comment by gzm0** [comment]
Updated

**Comment by sjrd** [comment]
LGTM

**Pull request merged**